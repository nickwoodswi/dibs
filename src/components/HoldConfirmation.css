.hold-confirmation{
    display: flex;
    flex-wrap: wrap;
    background-color: rgb(0, 0, 0, .8);
    color: white;
    max-width: 600px;
    margin-left: auto;
    margin-right: auto;
    padding: 15px;
    margin-bottom: 25px;
}
    .hold-confirmation-ack h2{
        text-align: center;
    }
    .hold-confirmation-ack h3{
        text-align: left;
    }
    .date-order{
        margin-left: auto;
        margin-right: auto;
    }