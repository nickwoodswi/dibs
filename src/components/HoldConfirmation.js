import React, {Component} from 'react'
import ConfirmationComponent from './ConfirmationComponent'
import './HoldConfirmation.css'

class HoldConfirmation extends Component {
    render() {
        if (!this.props.holds_added) {
            return <h1>Waiting to add holds...</h1>
        }
        else {
            return(
                <div className="hold-confirmation">
                    <div className="hold-confirmation-ack">
                        <h2><i>THANKS FOR PLACING HOLDS AT OUR VENUE!</i></h2>
                        <h3>We will respond with questions, concerns, and/or an offer within the next 48 hours, or within 4 months of your hold dates - whichever comes later. <br/>
                        <br/>
                        <i>Your order for the following dates is currently:</i> </h3>
                    </div>
                    <div className="date-order">
                        {this.props.holds_added.map(hold => {
                            return(
                                <ConfirmationComponent 
                                    holdDate={Object.keys(hold)}
                                    holdOrder={Object.values(hold)} />
                            )
                        })}
                    </div>
                    <div className="follow-up">
                        <p>Please feel free to follow up with Nick Woods at <a href="mailto:nick.woods.wi@gmail.com">nick.woods.wi@gmail.com</a> to challenge existing holds above yours, release your holds, or confirm an offer. We use DIBS to track holds on our calendar, and will notify you of any status changes to any of the holds you just placed.</p>
                    </div>
                </div>
            )
        }
    }
}

export default HoldConfirmation