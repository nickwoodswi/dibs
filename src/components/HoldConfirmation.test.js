import React from 'react';
import { render } from '@testing-library/react';
import HoldConfirmation from './HoldConfirmation';
import { MemoryRouter } from 'react-router-dom'
import renderer from 'react-test-renderer'

test('renders hold confirmation component', () => {
    render(<MemoryRouter><HoldConfirmation /></MemoryRouter>);
});

it('HoldConfirmation renders the UI as expected', () => {
    const tree = renderer
        .create(<MemoryRouter><HoldConfirmation /></MemoryRouter>)
        .toJSON();
    expect(tree).toMatchSnapshot();
});