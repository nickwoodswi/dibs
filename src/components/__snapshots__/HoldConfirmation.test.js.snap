// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HoldConfirmation renders the UI as expected 1`] = `
<h1>
  Waiting to add holds...
</h1>
`;
